common/robotPositioner.h

```diff
@@ -0,0 +1,186 @@
+
+    RobotPositioner(
+        BoBRobotics::Robots::Tank &tank,
+        millimeter_t robot_r,                                         // robot's wheel radius
+        millimeter_t robot_D,                                         // robots' axis length
+        millimeter_t stopping_distance,                               // if the robot's distance from goal < stopping dist, robot stops
+        degree_t allowed_heading_error,                               // the amount of error allowed in the final heading
+        double k1,                                                    // curveness of the path to the goal
+        double k2,                                                    // speed of turning on the curves
+        double alpha,                                                 // causes more sharply peaked curves 
+        double beta,                                                  // causes to drop velocity if 'k'(curveness) increases
+        meters_per_second_t max_velocity                              // max velocity
+        ) : m_bot(tank),
+            m_robot_r(robot_r),
+            m_robot_D(robot_D),
+            m_stopping_distance(stopping_distance),
+            m_allowed_heading_error(allowed_heading_error),
+            m_k1(k1),
+            m_k2(k2),
+            m_alpha(alpha),
+            m_beta(beta),
+            m_max_velocity(max_velocity) 
+    {/* setup */ }
+
+    //! sets the goal pose (x, y, angle)
+    void setGoalPose(millimeter_t pos_x, millimeter_t pos_y, degree_t goal_angle) {
+        m_goalPositionX = pos_x;
+        m_goalPositionY = pos_y;
+        m_goalAngle = goal_angle;
+    }
+
+    
+
+    //! updates the motors so the robot drives towards the goal location 
+    void updateMotors(millimeter_t pos_x, millimeter_t pos_y, degree_t angle) {
+        m_pos_X = pos_x;
+        m_pos_Y = pos_y;
+        m_heading = angle;
+        
+        // updates the range and bearing
+        updateRangeAndBearing();
+
+        // orientation of Target with respect to the line of sight from the observer to the target
+        degree_t theta = m_heading + m_bearingFromGoal - m_goalAngle; 
+        theta = angleWrapAround(theta);
+
+        degree_t delta = units::math::atan(scalar_t( (-m_k1 * theta).value()));
+        degree_t part1 = m_k2 * (m_bearingFromGoal-delta);
+        degree_t part2 = (1_deg + degree_t((m_k1/(1_sq_deg + 
+            units::math::pow<2>(m_k1*theta))).value()))* units::math::sin(m_bearingFromGoal);
+
+        auto k = (-(part1+part2)/m_distanceFromGoal);
+
+        meters_per_second_t v = m_max_velocity/scalar_t( (1+m_beta*pow(abs(k.value()),m_alpha)));
+        degrees_per_second_t w = k*v;
+
+        driveRobotWithVelocities(v,w, m_robot_r, m_robot_D);  
+    }
+
+    //! returns true if the robot reached the goal position
+    bool didReachGoal() {
+        return (m_distanceFromGoal < m_stopping_distance && 
+            closeTo(m_bearingFromGoal,m_allowed_heading_error));
```

Comment:
as closeTo is only used once it doesn't make a lot of sense having it as a function - just stick the code in here